Issues found when building Chakra with MSVC + /permissive-

Building Chakra with MSVC + /permissive- failed with error C4596, the issue is in  https://github.com/Microsoft/ChakraCore/blob/8d8cfe53bb111797ebdeb02c3343c1b9a53da067/lib/Runtime/Library/JavascriptLibrary.h#L1366.

Using full-qualified name to declare members inside class is ill-formed and incorrectly allowed by MSVC. MSVC now gives error when /permissive- is used. The fix is remove ‘JavascriptLibrary::’

Failure like:
d:\chakra\src\lib\runtime\Library/JavascriptLibrary.h(1366): error C4596: 'DefaultCreateFunction': illegal qualified name in member declaration

--- a/lib/Runtime/Library/JavascriptLibrary.h
+++ b/lib/Runtime/Library/JavascriptLibrary.h
@@ -1364,7 +1364,7 @@ namespace Js
         static bool __cdecl InitializeAsyncFunction(DynamicObject *function, DeferredTypeHandlerBase * typeHandler, DeferredInitializeMode mode);
 
         RuntimeFunction* CreateBuiltinConstructor(FunctionInfo * functionInfo, DynamicTypeHandler * typeHandler, DynamicObject* prototype = nullptr);
-        void JavascriptLibrary::DefaultCreateFunction(ParseableFunctionInfo * functionInfo, int length, DynamicObject * prototype, PropertyId nameId);
+        void DefaultCreateFunction(ParseableFunctionInfo * functionInfo, int length, DynamicObject * prototype, PropertyId nameId);
         RuntimeFunction* DefaultCreateFunction(FunctionInfo * functionInfo, int length, DynamicObject * prototype, DynamicType * functionType, PropertyId nameId);
         RuntimeFunction* DefaultCreateFunction(FunctionInfo * functionInfo, int length, DynamicObject * prototype, DynamicType * functionType, Var nameId);
         JavascriptFunction* AddFunction(DynamicObject* object, PropertyId propertyId, RuntimeFunction* function);

